subprojects/gstreamer/plugins/tracers/gstframerate.c

```diff
@@ -134,9 +355,23 @@ gst_framerate_tracer_display_current_fps (GstFramerateTracer *self,
   if (dr == 0.0) {
     gst_tracer_record_log (tr_framerate, self->pid, current_ts, frames_rendered,
         frames_dropped, average_fps, -1.0f, -1.0f);
+    g_snprintf (data, sizeof (data), CLIENT_SOCKET_DATA_FORMAT,
+        self->pid, current_ts, frames_rendered, frames_dropped, average_fps,
+        -1.0f, -1.0f);
   } else {
     gst_tracer_record_log (tr_framerate, self->pid, current_ts, frames_rendered,
         frames_dropped, -1.0f, rr, dr);
+    g_snprintf (data, sizeof (data), CLIENT_SOCKET_DATA_FORMAT,
+        self->pid, current_ts, frames_rendered, frames_dropped, -1.0f, rr, dr);
+  }
+
+  if (self->unix_socket_path != NULL) {
+    send_data = g_new0 (GstFramerateClientSocketSendData, 1);
+    send_data->self = self;
+    send_data->data = g_strdup (data);
+    send_data->len = sizeof (data);
+    g_thread_new (NULL, (GThreadFunc) gst_framerate_client_send_on_thread,
```

Comment:
in a sake of correctness of programming this thread has to be joined somewhere. Maybe it would be better to call `gst_element_call_async` instead.

Reply:
But the tracer is not an element, or should I pass the sink element here?

Reply:
Maybe I should join them on tracer destruction? But there will be a lot of threads :thinking:

Reply:
Imho since the socket is only one the thread should also be the only one. Also to avoid sending multiple messages in parallel.

Reply:
it is ok